app/src/migration/java/org/mozilla/fenix/MigrationPushRenewer.kt

```diff
@@ -1,40 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-package org.mozilla.fenix
-
-import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.flow.collect
-import mozilla.components.concept.push.PushProcessor
-import mozilla.components.lib.state.ext.flowScoped
-import mozilla.components.support.base.log.logger.Logger
-import mozilla.components.support.migration.state.MigrationProgress
-import mozilla.components.support.migration.state.MigrationStore
-
-/**
- * Force-renews push subscription after migration was complete.
- */
-class MigrationPushRenewer(
-    private val service: PushProcessor?,
-    private val store: MigrationStore
-) {
-
-    @OptIn(ExperimentalCoroutinesApi::class)
-    fun start() {
-        // Observe for migration completed.
-        store.flowScoped { flow ->
-            flow.collect { state ->
-                Logger("MigrationPushRenewer").debug("Migration state: ${state.progress}")
-                if (state.progress == MigrationProgress.COMPLETED) {
-                    Logger("MigrationPushRenewer").debug("Renewing registration....")
-
-                    // This should force a recreation of firebase device token, re-registration with
-                    // the autopush service, and subsequent update of the FxA device record with
-                    // new push subscription information.
-                    service?.renewRegistration()
```

Comment:
@jonalmeida What will be the side effect of not running this for a Fenix users on upgrade? Will they ever recover from this or will they be stuck forever?

Reply:
TL;DR: We should be fine for Sync and web content subscriptions may be able to recover on their own.

There are three things here to consider: GCM/Firebase push token, WebPush subscriptions, AutoPush keys

When migrating, away from GCM, Firebase servers knows how to handle the old GCM token and tell FCM clients to request a new token. A new token is then submitted to AutoPush so our servers are aware of the token change. So that covers the two cases.

For existing subscriptions, AutoPush would need to tell us to request new subscriptions or another means. With Sync, if the Sync servers cannot communicate to the device, a future sync from polling would tell the client that push is not working which will trigger the renew process. However if the user doesn't have Sync, then browser clients will not know to renew their subscriptions, but this something that web clients usually should re-request. With mhammond's recent patches for push, this should make the process better along with other [future changes](https://docs.google.com/document/d/11mUtkK6lzSNuzjoYw1K0lDBRGuACbfTIHIXSDibX540/edit#).